hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountChangeValidator.java

```diff
@@ -0,0 +1,125 @@
+package com.netflix.hollow.api.producer.validation;
+
+import com.netflix.hollow.api.consumer.data.AbstractHollowDataAccessor;
+import com.netflix.hollow.api.producer.HollowProducer;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import com.netflix.hollow.core.read.engine.HollowTypeReadState;
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class RecordCountChangeValidator implements ValidatorListener {
+    private static final String NAME = RecordCountVarianceValidator.class.getName();
+    private final Logger log = Logger.getLogger(RecordCountChangeValidator.class.getName());
+
+    private final String typeName;
+    private final Threshold threshold;
+    private AbstractHollowDataAccessor accessor;
+
+    public RecordCountChangeValidator(String typeName,
+                                      Threshold threshold) {
+        this.typeName = typeName;
+        this.threshold = threshold;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public ValidationResult onValidate(HollowProducer.ReadState readState) {
+        HollowReadStateEngine readStateEngine = requireNonNull(readState.getStateEngine(), "read state is null");
+        HollowTypeReadState typeState = requireNonNull(readStateEngine.getTypeState(typeName),
+                "type not loaded or does not exist in dataset; type=" + typeName);
+        accessor =
+            new AbstractHollowDataAccessor<Object>(readStateEngine, typeName) {
+                @Override public Object getRecord(int ordinal) { return null; }
+        };
+        ValidationResult validationResult = validateChanges(typeState);
+        log.info(validationResult.toString());
+        return validationResult;
+    }
+
+    private ValidationResult validateChanges(HollowTypeReadState typeState) {
+        if(typeState.getPreviousOrdinals().isEmpty()) {
+            return ValidationResult.from(this).passed("Ignore the check if previous records are empty.");
+        }
+        int addRecordNumber = accessor.getAddedRecords().size();
+        int removeRecordNumber = accessor.getRemovedRecords().size();
+        int updatedRecordNumber = accessor.getUpdatedRecords().size();
+        int previousRecordNumber = typeState.getPreviousOrdinals().cardinality();
+
+        float addedPercent = (float) addRecordNumber / previousRecordNumber;
+        float removedPercent = (float) removeRecordNumber / previousRecordNumber;
+        float updatedPercent = (float) updatedRecordNumber / previousRecordNumber;
+
+
+        float addedPercentageThreshold = threshold.addedPercentageThreshold;
```

Comment:
lets make these dynamically configurable for when producers want to let a publish through